src/script/miniscript.cpp

```diff
@@ -282,16 +280,15 @@ size_t ComputeScriptLen(Fragment nodetype, Type sub0typ, size_t subsize, uint32_
     return 0;
 }
 
-bool DecomposeScript(const CScript& script, std::vector<std::pair<opcodetype, std::vector<unsigned char>>>& out)
+std::optional<std::vector<std::pair<opcodetype, std::vector<unsigned char>>>> DecomposeScript(const CScript& script)
```

Comment:
I'm not a fan of typedef, but I think that one typedef, in this case, can make the return type cleaner from a reader's viewpoint?

```suggestion
// FIXME: move this declaration in some better place!
typedef std::pair<opcodetype, std::vector<unsigned char>> OPCode

std::optional<std::vector<OPCode>> DecomposeScript(const CScript& script)
```

Reply:
Done